pkg/registry/scripts/main.js

```diff
@@ -58,20 +60,17 @@
     ])
 
     .controller('MainCtrl', [
+        '$q',
         '$scope',
         '$location',
         '$rootScope',
         '$timeout',
         'kubeLoader',
-        'kubeSelect',
+        'registrySettings',
         'filterService',
         'cockpitKubeDiscover',
-        function($scope, $location, $rootScope, $timeout, loader, select, filter, discover) {
```

Comment:
The way i'm doing this in the big refactor is to not use discover here, but have the settings factory handle everything and just have the curtains display wait on that. https://github.com/petervo/cockpit/blob/kube-refactor/pkg/kubernetes/scripts/app.js

I'll need to rebase on this anyways, no need to change anything here we can discuss further in the other PR if you prefer.